homeassistant/components/kegtron/sensor.py

```diff
@@ -0,0 +1,136 @@
+    SensorUpdate,
+    Units,
+)
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataProcessor,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothProcessorCoordinator,
+    PassiveBluetoothProcessorEntity,
+)
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import SIGNAL_STRENGTH_DECIBELS_MILLIWATT, VOLUME_LITERS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .device import device_key_to_bluetooth_entity_key, sensor_device_info_to_hass
+
+SENSOR_DESCRIPTIONS = {
+    "port_count": SensorEntityDescription(
+        key="port_count",
+        icon="mdi:water-pump",
+    ),
+    "keg_size": SensorEntityDescription(
+        key="keg_size",
+        icon="mdi:keg",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "keg_type": SensorEntityDescription(
+        key="keg_type",
+        icon="mdi:keg",
+    ),
+    "volume_start": SensorEntityDescription(
+        key="volume_start",
+        icon="mdi:keg",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "volume_dispensed": SensorEntityDescription(
+        key="volume_dispensed",
+        icon="mdi:keg",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    "port_state": SensorEntityDescription(
+        key="port_state",
+        icon="mdi:water-pump",
+    ),
+    "port_name": SensorEntityDescription(
+        key="port_name",
+        icon="mdi:water-pump",
+    ),
+    "signal_strength": SensorEntityDescription(
+        key=f"{KegtronSensorDeviceClass.SIGNAL_STRENGTH}_{Units.SIGNAL_STRENGTH_DECIBELS_MILLIWATT}",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+
+def sensor_update_to_bluetooth_data_update(
+    sensor_update: SensorUpdate,
+) -> PassiveBluetoothDataUpdate:
+    """Convert a sensor update to a bluetooth data update."""
+    return PassiveBluetoothDataUpdate(
+        devices={
+            device_id: sensor_device_info_to_hass(device_info)
+            for device_id, device_info in sensor_update.devices.items()
+        },
+        entity_descriptions={
+            device_key_to_bluetooth_entity_key(device_key): SENSOR_DESCRIPTIONS[
+                description.device_key.key.removesuffix("_port_1").removesuffix(
+                    "_port_2"
+                )
```

Comment:
Is it possible to get rid of the `removesuffix`?

Reply:
It is possible, but that would most likely mean that I have to add the `DEVICE_CLASS__UNIT` combination into the `sensor-state-data` package. I was thinking that the Kegtron sensor types where not that common, so I left them out. Is it ok to put them into `sensor-state-data`?

Reply:
And if not `update_predefined_sensor` is just a wrapper around `update_sensor`
https://github.com/Bluetooth-Devices/sensor-state-data/blob/154cbdd33202d17a619c1af503d56400d4a93666/src/sensor_state_data/data.py#L220

Reply:
I tried it with the `device_id`, like you did in Govee, but I couldn't get that to work. It was creating entities without the _port_#, so I only got entities without `_port_x`. I was probably doing something wrong, but I gave up. 

The issue with `update_sensor` is that it requires a SensorDeviceClass, which is defined in sensor-state-data.sensor.device_class. Can't use a string there. 

![image](https://user-images.githubusercontent.com/24375847/191008752-2b97b652-2358-4a45-9fc4-f5d673807e25.png)

I guess the easiest solution is adding the sensor types to `sensor-state-data.sensor.device_class`

Reply:
> I guess the easiest solution is adding the sensor types to `sensor-state-data.sensor.device_class`

Without actually seeing the PR, I think that makes sense

Reply:
If you call `set_device_name` and `set_device_type` for that `device_id` it should work as expected

https://github.com/Bluetooth-Devices/govee-ble/blob/49a5424d3c316045b2ee24e3a0ef9553a2d8fd20/src/govee_ble/parser.py#L180

Reply:
Ah, thanks, that was what I was missing. If you can have a look at the [PR for sensor-stata-data](https://github.com/Bluetooth-Devices/sensor-state-data/pull/35), I can finish this this. Thanks again for pointing in the right direction!

Reply:
Fixed it by using the device_id